main.tf

```diff
@@ -1,11 +1,11 @@
 data "external" "this" {
-  count = "${length(var.uris)}"
+  for_each = toset(var.uris)
```

Comment:
just curious, but why convert this list to a set?

Reply:
At the resource level, `for_each` only works with maps or sets. That's because a list can have duplicate values, a set cannot. With `for_each` the value becomes the resource id, and cannot have duplicate resource ids...

Reply:
ah, good to know. thanks!